politeiawww/proposals.go

```diff
@@ -202,46 +202,47 @@ func validateProposal(np www.NewProposal, u *user.User) error {
 	// Check for at least 1 markdown file with a non-empty payload.
 	if len(np.Files) == 0 || np.Files[0].Payload == "" {
 		return www.UserError{
-			ErrorCode: www.ErrorStatusProposalMissingFiles,
+			ErrorCode: www.ErrorStatusNoIndexFile,
 		}
 	}
 
 	// verify if there are duplicate names
 	filenames := make(map[string]int, len(np.Files))
 	// Check that the file number policy is followed.
 	var (
-		numMDs, numImages, numIndexFiles      int
-		mdExceedsMaxSize, imageExceedsMaxSize bool
-		hashes                                []*[sha256.Size]byte
+		numAttachments, numIndexFiles int
+		indexExceedsMaxSize           bool
+		attachmentExceedsMaxSize      bool
+		hashes                        []*[sha256.Size]byte
 	)
 	for _, v := range np.Files {
 		filenames[v.Name]++
+
 		var (
 			data []byte
 			err  error
 		)
-		if strings.HasPrefix(v.MIME, "image/") {
-			numImages++
-			data, err = base64.StdEncoding.DecodeString(v.Payload)
-			if err != nil {
-				return err
-			}
-			if len(data) > www.PolicyMaxImageSize {
-				imageExceedsMaxSize = true
-			}
-		} else {
-			numMDs++
 
-			if v.Name == indexFile {
-				numIndexFiles++
-			}
+		data, err = base64.StdEncoding.DecodeString(v.Payload)
+		if err != nil {
+			return err
+		}
 
-			data, err = base64.StdEncoding.DecodeString(v.Payload)
-			if err != nil {
-				return err
+		if v.Name == indexFile {
+			numIndexFiles++
+			if len(data) > www.PolicyMaxIndexFileSize {
+				indexExceedsMaxSize = true
+			}
+			// Enforces that index file is in fact a markdown file
+			if v.MIME != "text/plain; charset=utf-8" {
+				return www.UserError{
+					ErrorCode: www.ErrorStatusInvalidMIMEType,
```

Comment:
There should be different errors for invalid index mime type and invalid attachment mime types.